enzyme/test/Integration/CppSugar/dup_args.cpp

```diff
@@ -0,0 +1,121 @@
+// RUN: if [ %llvmver -ge 11 ]; then %clang++ -std=c++17 -O0 %s -S -emit-llvm -o - %loadClangEnzyme | %lli - ; fi
+// RUN: if [ %llvmver -ge 11 ]; then %clang++ -std=c++17 -O1 %s -S -emit-llvm -o - %loadClangEnzyme | %lli - ; fi
+// RUN: if [ %llvmver -ge 11 ]; then %clang++ -std=c++17 -O2 %s -S -emit-llvm -o - %loadClangEnzyme | %lli - ; fi
+// RUN: if [ %llvmver -ge 11 ]; then %clang++ -std=c++17 -O3 %s -S -emit-llvm -o - %loadClangEnzyme | %lli - ; fi
+// RUN: if [ %llvmver -ge 12 ]; then %clang++ -std=c++17 -O0 %s -S -emit-llvm -o - %newLoadClangEnzyme | %lli - ; fi
+// RUN: if [ %llvmver -ge 12 ]; then %clang++ -std=c++17 -O1 %s -S -emit-llvm -o - %newLoadClangEnzyme | %lli - ; fi
+// RUN: if [ %llvmver -ge 12 ]; then %clang++ -std=c++17 -O2 %s -S -emit-llvm -o - %newLoadClangEnzyme | %lli - ; fi
+// RUN: if [ %llvmver -ge 12 ]; then %clang++ -std=c++17 -O3 %s -S -emit-llvm -o - %newLoadClangEnzyme | %lli - ; fi
+
+#include "../test_utils.h"
+
+#include <enzyme/enzyme>
+
+float f(float x)        { return  x *  x; }
+float g(const float& x) { return  x *  x; }
+float h(const float* x) { return *x * *x; }
+
+int main() {
+    const float x      = 10;
+    const float etalon = 2 * x;
+
+    {   // Ok, reverse mode + active
+        const float dfdx = enzyme::get<0>(
+            enzyme::get<0>(
+                enzyme::autodiff<enzyme::Reverse>(
+                    f, enzyme::Active{ x }
+                )
+            )
+        );
+        APPROX_EQ(dfdx, etalon, 1e-10);
+    }
+#if 0
+    {
+        /*
+         * This will not work in the C++ interface because it is also not allowed
+         * with the C-style __enzyme_autodiff call
+         */
+        const float dfdx = enzyme::get<0>(
+            enzyme::get<0>(
+                enzyme::autodiff<enzyme::Reverse>(
+                    g, enzyme::Active<const float&>{ x }
+                )
+            )
+        );
+        APPROX_EQ(dfdx, etalon, 1e-10);
+    }
+#endif
+#if 0
+    {
```

Comment:
We should be able to make a separate file and do verify errors here. See this for example: https://github.com/EnzymeAD/Enzyme/blob/main/enzyme/test/Integration/ReverseMode/err_empty.c